packages/components/src/components/drawer/drawer.scss

```
@@ -0,0 +1,171 @@
+@use "./../../styles/dialog-init" as *;
+@use "@db-ui/foundations/build/scss/variables" as *;
+@use "@db-ui/foundations/build/scss/variables.global" as *;
+@use "@db-ui/foundations/build/scss/helpers/functions" as *;
+@use "@db-ui/foundations/build/scss/helpers/component" as *;
+@use "@db-ui/foundations/build/scss/animation/animations" as *;
+
+%direction-right {
+	left: auto;
+	right: 0;
+
+	&[data-rounded="true"] {
+		border-top-left-radius: $default-card-border-radius;
+		border-bottom-left-radius: $default-card-border-radius;
+	}
+}
+
+%direction-left {
+	right: auto;
+	left: 0;
+
+	&[data-rounded="true"] {
+		border-top-right-radius: $default-card-border-radius;
+		border-bottom-right-radius: $default-card-border-radius;
+	}
+}
+
+%full-horizontal {
+	&:not([data-width="full"]) {
+		max-height: var(
+			--db-drawer-max-height,
+			calc(100% - $db-spacing-fixed-xl)
+		);
+		max-width: none;
+	}
+}
+
+%direction-up {
+	@extend %full-horizontal;
+	top: auto;
+	bottom: 0;
+	left: 0;
+	right: 0;
+
+	&[data-rounded="true"] {
+		border-top-right-radius: $default-card-border-radius;
+		border-top-left-radius: $default-card-border-radius;
+	}
+}
+%direction-down {
+	@extend %full-horizontal;
+	bottom: auto;
+	top: 0;
+	left: 0;
+	right: 0;
+
+	&[data-rounded="true"] {
+		border-bottom-right-radius: $default-card-border-radius;
+		border-bottom-left-radius: $default-card-border-radius;
+	}
+}
+
+.db-drawer {
+	.db-drawer-container {
+		@extend %direction-right;
+		display: flex;
+		flex-direction: column;
+		background-color: $db-colors-neutral-bg-0-enabled;
+		padding-block: $db-spacing-fixed-sm;
+		height: 100%;
+		width: 100%;
+		position: fixed;
+
+		&:not([data-width="full"]) {
+			max-width: var(
+				--db-drawer-max-width,
+				calc(100% - $db-spacing-fixed-xl)
+			);
+		}
+
+		&[data-size="small"] {
+			padding-block: $db-spacing-fixed-xs;
+
+			.db-drawer-header {
+				padding-block-end: $db-spacing-fixed-xs;
+				padding-inline: $db-spacing-fixed-md;
+			}
+
+			.db-drawer-content {
+				padding-inline: $db-spacing-fixed-md;
+			}
+		}
+
+		&[data-rounded="true"] {
+			box-shadow: $db-elevation-4;
+		}
+
+		&[data-direction="left"] {
```

```suggestion
		&[data-direction="start"] {
```

I'm struggeling with non-semantic positional values – but even also `start` and `end` don't seem to perfectly fit here. What about `forth` and `back`?

---

I checked other design-systems: https://mui.com/material-ui/react-drawer/#swipeable

I think left and right, up and down makes sense and it's simple. We could name it `from-right-to-left` to make it bulletproof but it could be a little bit of an overkill

---

let's stay better than mui.